src/lib/cat_stories/index.svelte

```diff
@@ -49,45 +55,194 @@
     openOverlayIndex = -1;
   }
 
-  function openCatOverlay(index) {
+  function openCatOverlay(index: number) {
     openOverlayIndex = index;
   }
+
+  let lastScrollLeft: number | null = null;
+  let lastScrollTimestampMS = 0;
+  let snapTimeout = null;
+  let periodicTimeout = null;
+  let periodicScrollLeft: number | null = null;
+
+  function scrollIfNotMoving(scrollCB: () => void) {
+    const storiesContainer = document.querySelector('.stories-container');
+    if (
+      periodicScrollLeft !== null &&
+      storiesContainer !== null &&
+      storiesContainer.scrollLeft === periodicScrollLeft
+    ) {
+      clearTimeout(periodicTimeout);
+      scrollCB();
+    } else {
+      periodicTimeout = setTimeout(() => scrollIfNotMoving(scrollCB), 50);
+    }
+    periodicScrollLeft = storiesContainer.scrollLeft;
+  }
+
+  function getNodeTransform(node: HTMLElement, center: number) {
+    const rect = node.getBoundingClientRect();
+    const val = Math.pow(
+      Math.cos(((Math.PI / 2) * (rect.left + rect.width / 2 - center) * 0.4) / center),
+      center <= 250 ? 1 : 2
+    );
+    const clamp = Math.max(0.75, Math.min(1, val));
+    return clamp;
+  }
+
+  function scaleChildren(childNodes, center) {
+    childNodes.forEach((_childNode) => {
+      const childNode = _childNode as HTMLElement;
+      if (childNode.childNodes.length > 0 && (childNode.childNodes[0] as HTMLElement).style) {
+        (childNode.childNodes[0] as HTMLElement).style.transform = 'scale(1)';
+        childNode.offsetHeight;
+        const clamp = getNodeTransform(childNode, center);
+        (childNode.childNodes[0] as HTMLElement).style.transform = `scale(${clamp})`;
+      }
+    });
+  }
+
+  function parseTransform(transformStr: string) {
```

Comment:
I think instead of this function, it might be worth creating an array and modifying it in `getNodeTransform` (then accessing it when we would have called this function).

Regex matching is very slow, so anything we can do to avoid it is ideal (especially an `O(1)` operation like writing to an array).